### PR TITLE
feat(tables): Selection

### LINKED ISSUE #8027: [tables] Selection (checkbox column)

### PR DESCRIPTION
**Pull request by ZaymonFC**
- Resolves #8027 
- Supports bulk deletion
- Selection is durable through sort and row order changes, insertions and deletions etc.

https://github.com/user-attachments/assets/b279a2cf-a49f-4953-9e9e-b1a787f6f0f9

### CODE REVIEW

File: packages/plugins/plugin-space/src/meta.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ export enum SpaceAction {
   CLOSE = `${SPACE_ACTION}/close`,
   MIGRATE = `${SPACE_ACTION}/migrate`,
   ADD_OBJECT = `${SPACE_ACTION}/add-object`,
```
Comment (wittjosiah) [inline packages/plugins/plugin-space/src/meta.ts:27]: todo: add objects

File: packages/plugins/plugin-space/src/translations.ts [hunk 2]
```diff
@@ -47,6 +47,7 @@ export default [
         'delete object label': 'Delete item',
         'collection deleted label': 'Collection deleted',
         'object deleted label': 'Item deleted',
+        'objects deleted label': 'Items deleted',
```
Comment (wittjosiah) [inline packages/plugins/plugin-space/src/translations.ts:50]: see https://github.com/dxos/dxos/blob/da7e0b3e001c14e9c78cb47add1c95ab01a5c651/packages/plugins/plugin-space/src/translations.ts#L70-L72

Comment (ZaymonFC) [reply packages/plugins/plugin-space/src/translations.ts:50]: Hmm, none of the `{{count}}` translations I could find seem to have usages. How do I supply the params?

Comment (wittjosiah) [reply packages/plugins/plugin-space/src/translations.ts:50]: https://www.i18next.com/translation-function/plurals

Comment (ZaymonFC) [reply packages/plugins/plugin-space/src/translations.ts:50]: https://github.com/dxos/dxos/blob/a597fa5190f0fe057687eecb8d5bfea610eda224/packages/plugins/plugin-space/src/SpacePlugin.tsx#L1347-L1359

Hmmm.

Comment (wittjosiah) [reply packages/plugins/plugin-space/src/translations.ts:50]: Ah yeah that would be problematic, just TODO I guess

**Pull request merged**